jaeger-core/src/main/java/io/jaegertracing/internal/propagation/HexCodec.java

```diff
@@ -32,7 +32,21 @@ static Long lowerHexToUnsignedLong(String lowerHex) {
     // trim off any high bits
     int beginIndex = length > 16 ? length - 16 : 0;
 
-    return lowerHexToUnsignedLong(lowerHex, beginIndex);
+    return hexToUnsignedLong(lowerHex, beginIndex, Math.min(beginIndex + 16, lowerHex.length()));
+  }
+
+  /**
+   * Parses a 1 to 32 character higher-hex string with no prefix into an unsigned long, tossing any
+   * bits higher than 64.
+   *
+   * @return a 64 bit long, meaning that negative values are the overflow of Java's 32 bit long
+   */
+  static Long higherHexToUnsignedLong(String higherHex) {
+    if (higherHex.length() > 32 || higherHex.length() < 1) {
+      return null;
```

Comment:
I think it would be worth a log message somewhere about this...

Reply:
Done at debug Level. I though warn would be also good since it's a situation that should arrive rarely or that can cause tracing to malfunction.
Let me know what you think.